src/core/pubsub.ts

```diff
@@ -0,0 +1,16 @@
+import type { _EventMap } from './gosling.schema';
+
+export function publish<Name extends keyof _EventMap>(name: Name, data: _EventMap[Name]): void {
+    PubSub.publish(name, data);
+}
+
+export function subscribe<Name extends keyof _EventMap>(
+    name: Name,
+    callback: (msg: string, data: _EventMap[Name]) => void
+): void {
+    PubSub.subscribe(name, callback);
+}
+
+export function unsubscribe<Name extends keyof _EventMap>(name: Name): void {
```

Comment:
This can just be

```suggestion
export function unsubscribe(name: keyof _EventMap): void {
Write
```

since the generic type parameter does need to be used in the function signature.

Reply:
Better yet, a local type alias

```typescript
type EventName = keyof _EventMap;
```

might be nice in this file.

Reply:
I agree!